taf/updater/updater.py

```diff
@@ -68,6 +68,39 @@ def _check_update_status(repos_update_data, auth_repo_name, host_update_status,
     return update_update_status, errors
 
 
+def _clone_validation_repo(url, repository_name, default_branch):
+    """
+    Clones the authentication repository based on the url specified using the
+    mirrors parameter. The repository is cloned as a bare repository
+    to a the temp directory and will be deleted one the update is done.
+
+    If repository_name isn't provided (default value), extract it from info.json.
+    """
+    temp_dir = tempfile.mkdtemp()
+    path = Path(temp_dir, "auth_repo").absolute()
+    validation_auth_repo = GitRepository(path=path, urls=[url])
+    validation_auth_repo.clone(bare=True)
+    validation_auth_repo.fetch(fetch_all=True)
+
+    settings.validation_repo_path = validation_auth_repo.path
+
+    validation_head_sha = validation_auth_repo.top_commit_of_branch(default_branch)
+
+    if repository_name is None:
+        try:
+            info = validation_auth_repo.get_json(
+                validation_head_sha, "targets/protected/info.json"
```

Comment:
info.json path should be a constant. At some point, we should move all of the paths that the framework needs to the same place